### PR TITLE
Avoid duplicate scopes in cached privileges

### LINKED ISSUE #661: Duplicate scopes in privilege cache

### PR DESCRIPTION
**Pull request by xcoulon**
list scopes using the `DISTINCT` SQL option to avoid duplicates
added tests

also, fixed a few errors reported when building with go 1.11

fixes #661

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-663-1`

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 1]
```diff
@@ -132,10 +133,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: can't we remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 2]
```diff
@@ -132,10 +133,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: can't we remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go (hunk 1 above)
Comment (xcoulon) [reply authorization/invitation/service/invitation_service_blackbox_test.go]: sure, let me do that

Comment (xcoulon) [reply authorization/invitation/service/invitation_service_blackbox_test.go]: @dipak-pawar done in 3f65439

File: authorization/invitation/service/invitation_service_blackbox_test.go (hunk 2 above)
Comment (xcoulon) [reply authorization/invitation/service/invitation_service_blackbox_test.go]: done in 3f65439

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-663-2`

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 3]
```diff
@@ -267,10 +266,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 4]
```diff
@@ -267,10 +266,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 5]
```diff
@@ -317,10 +316,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 6]
```diff
@@ -317,10 +316,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 7]
```diff
@@ -358,10 +357,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 8]
```diff
@@ -358,10 +357,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 9]
```diff
@@ -397,10 +396,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 10]
```diff
@@ -397,10 +396,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 11]
```diff
@@ -434,10 +433,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
 			},
 		}
 
-		var messages []notification.Message
+		// var messages []notification.Message
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: authorization/invitation/service/invitation_service_blackbox_test.go [hunk 12]
```diff
@@ -434,10 +433,10 @@ func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
...
 		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(t)
 		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, msgs []notification.Message, p2 ...configuration.HTTPClientOption) (r chan error, r1 error) {
-			messages = msgs
+			// messages = msgs
```
Comment (dipak-pawar) [inline authorization/invitation/service/invitation_service_blackbox_test.go]: Can you please remove this commented code?

File: test/graph/identity_role_wrapper.go [hunk 13]
```diff
@@ -80,3 +82,7 @@ func newIdentityRoleWrapper(g *TestGraph, params []interface{}) interface{} {
 func (w *identityRoleWrapper) IdentityRole() *rolePkg.IdentityRole {
 	return w.identityRole
 }
+
+func (w *identityRoleWrapper) Delete() {
+	w.graph.app.IdentityRoleRepository().Delete(context.Background(), w.identityRole.IdentityRoleID)
```
Comment (sbryzak) [inline test/graph/identity_role_wrapper.go]: The graph has a `context` object you can use, via `w.graph.ctx`.

Comment (xcoulon) [reply test/graph/identity_role_wrapper.go]: ok, I'll change that. Thanks!

Comment (xcoulon) [reply test/graph/identity_role_wrapper.go]: addressed in https://github.com/fabric8-services/fabric8-auth/pull/663/commits/9532179f8f23a606f0c99e42c5451f13a9f6a87c

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-663-4`

**Pull request merged**